File: Baekjoon/Class4/02.15652 N과 M (4).js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim();

const [n, m] = input.split(" ").map(Number);
const track = [];
const out = [];

const backTrack = (start) => {
    if(track.length === m) { // 출력
        out.push(track.join(" "));
        return;
    }
    for(let i = start; i <= n; i++) {
        track.push(i); // 추가
        backTrack(i);
        track.pop(); // 삭제
    }
}

backTrack(1);
console.log(out.join("\n"));

File: Baekjoon/Class4/02.15652 N과 M (4).py
import sys
input = sys.stdin.readline
write = sys.stdout.write

n, m = map(int, input().split())
track = []

def back_track(start):
    if len(track) == m: #출력
        write(f"{" ".join(track)}\n")
        return
    for i in range(start, n + 1):
        track.append(str(i)) #추가
        back_track(i)
        track.pop() #삭제

back_track(1)